keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,527 @@
+
+Currently the noderesources scheduler plugin does not filter out the best-effort pods in the case there's no available CPU.
+
+Another option is to force the cpu requests for best effort pods to 1 MilliCPU in kubelet for the purpose of resource availability checks (or, equivalently, check there's at least 1 MilliCPU allocatable). This option is meant to be simpler than option-1, but it can create runaway pods similar to that in https://github.com/kubernetes/kubernetes/issues/84869.
+
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/policy_static.go`: `03-18-2024` - `91.1`
+
+##### Integration tests
+
+No new integration tests for kubelet are planned.
+
+##### e2e tests
+
+- These cases will be added in the existing e2e tests:
+  - CPU Manager works with `strict-cpu-reservation` policy option
+
+- Basic functionality
+1. Enable `CPUManagerPolicyAlphaOptions` feature gate and `strict-cpu-reservation` policy option.
+2. Create a simple pod of Burstable QoS type.
+3. Verify the pod is not using the reserved CPU cores.
+4. Delete the pod.
+
+
+### Graduation Criteria
+
+#### Alpha
+
+- [X] Implement the new policy option.
+- [X] Ensure proper unit tests are in place.
+
+#### Beta
+
+- [ ] Gather feedback from consumers of the new policy option.
+- [ ] Verify no major bugs reported in the previous cycle.
+
+#### GA
+
+- [ ] Allow time for feedback (1 year).
+- [ ] Make sure all risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+The new policy option is opt-in and orthogonal to the existing ones.
+
+### Version Skew Strategy
+
+No changes needed.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+The /var/lib/kubelet/cpu\_manager\_state needs be removed when enabling or disabling the feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyAlphaOptions`
+  - Components depending on the feature gate: `kubelet`
+- [X] Change the kubelet configuration to set a `CPUManager` policy of `static` and a `CPUManager` policy option of `strict-cpu-reservation`
+  - Will enabling / disabling the feature require downtime of the control plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning of a node?  Yes -- removing /var/lib/kubelet/cpu\_manager\_state and restarting kubelet are required.
```

Comment:
If the only requirement is to remove a file, I don't think you need to reprovision a full node, do you? I mean you can always do that, but it's not a hard requirement, right?

Reply:
yes, my original answer is "No", then I start to copy/paste other similar Kep's answers ... I will go back to No.